code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,335 @@
 	owner.heal_limb_damage(0, 5)
 	return ..()
 
+// ***************************************
+// *********** Essence Link
+// ***************************************
+/obj/effect/ebeam/essence_link
+	name = "essence link beam"
+
+/datum/status_effect/stacking/essence_link
+	id = "xeno_essence_link"
+	stacks = 0
+	stack_decay = -1 //Not meant to decay.
+	max_stacks = 3
+	consumed_on_threshold = FALSE
+	alert_type = null
+	/// The owner of the link.
+	var/mob/living/carbon/xenomorph/link_owner
+	/// Whom the owner is linked to.
+	var/mob/living/carbon/xenomorph/link_target
+	/// References the Essence Link action and its vars.
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action
+	/// If the target xeno was within range.
+	var/was_within_range = TRUE
+	/// Time it takes for the attunement levels to increase.
+	var/attunement_cooldown = 20 SECONDS
```

Comment:
If you have access to the action ref, why are these kind of balance var on the effect and not the action?

Reply:
Fair enough, done.